drivers/nrf24l01p_ng/nrf24l01p_ng_netdev.c

```diff
@@ -0,0 +1,648 @@
+#include <assert.h>
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#include "iolist.h"
+#include "irq.h"
+#include "luid.h"
+#include "mutex.h"
+#include "net/eui64.h"
+#include "net/netdev.h"
+#include "xtimer.h"
+
+#include "gnrc_netif_nrf24l01p_ng.h"
+#include "nrf24l01p_ng_constants.h"
+#include "nrf24l01p_ng_registers.h"
+#include "nrf24l01p_ng_communication.h"
+#include "nrf24l01p_ng_states.h"
+#include "nrf24l01p_ng_diagnostics.h"
+#include "nrf24l01p_ng_isr.h"
+#include "nrf24l01p_ng_netdev.h"
+
+#define NRF24L01P_NG_FLG_IRQ \
+    (NRF24L01P_NG_FLG_MAX_RT | NRF24L01P_NG_FLG_TX_DS | NRF24L01P_NG_FLG_RX_DR)
+
+static
+int _init(netdev_t *netdev);
+static
+int _recv(netdev_t *netdev, void *buf, size_t len, void *info);
+static
+int _send(netdev_t *netdev, const iolist_t *iolist);
+static
+void _isr(netdev_t *netdev);
+static
+int _get(netdev_t *netdev, netopt_t opt, void *val, size_t max_len);
+static
+int _set(netdev_t *netdev, netopt_t opt, const void *val, size_t max_len);
+
+const netdev_driver_t nrf24l01p_ng_driver = {
+    .init = _init,
+    .recv = _recv,
+    .send = _send,
+    .isr = _isr,
+    .get = _get,
+    .set = _set
+};
+
+static inline
+void _trigger_send(const nrf24l01p_ng_t *dev)
+{
+    gpio_set(dev->params.pin_ce);
+    xtimer_usleep(NRF24L01P_NG_DELAY_US_CE_HIGH_PULSE);
+    gpio_clear(dev->params.pin_ce);
+    xtimer_usleep(NRF24L01P_NG_DELAY_US_TX_SETTLING);
+}
+
+static
+int _assert_awake(const nrf24l01p_ng_t *dev)
+{
+    uint8_t config = 0;
+    nrf24l01p_ng_read_reg(dev, NRF24L01P_NG_REG_CONFIG, &config, 1);
+    return config & NRF24L01P_NG_FLG_PWR_UP;
+}
+
+static
+netopt_state_t _state_to_netif(nrf24l01p_ng_state_t state)
+{
+    if (state == NRF24L01P_NG_STATE_POWER_DOWN) {
+        return NETOPT_STATE_SLEEP;
+    }
+    if (state == NRF24L01P_NG_STATE_STANDBY_1) {
+        return NETOPT_STATE_STANDBY;
+    }
+    if (state == NRF24L01P_NG_STATE_STANDBY_2) {
+        return NETOPT_STATE_TX;
+    }
+    if (state == NRF24L01P_NG_STATE_TX_MODE) {
+        return NETOPT_STATE_TX;
+    }
+    if (state == NRF24L01P_NG_STATE_RX_MODE) {
+        return NETOPT_STATE_RX;
+    }
+    return NETOPT_STATE_OFF; /* error */
+}
+
+nrf24l01p_ng_state_t _state_from_netif(netopt_state_t state)
+{
+    if (state == NETOPT_STATE_SLEEP) {
+        return NRF24L01P_NG_STATE_POWER_DOWN;
+    }
+    if (state == NETOPT_STATE_STANDBY) {
+        return NRF24L01P_NG_STATE_STANDBY_1;
+    }
+    if (state == NETOPT_STATE_TX) {
+        return NRF24L01P_NG_STATE_TX_MODE;
+    }
+    if (state == NETOPT_STATE_RX) {
+        return NRF24L01P_NG_STATE_RX_MODE;
+    }
+    return NRF24L01P_NG_STATE_UNDEFINED;
+}
+
+static
+void nrf24l01p_ng_irq_handler(void *_dev)
+{
+    nrf24l01p_ng_t *dev = (nrf24l01p_ng_t *)_dev;
+    /* Once the IRQ pin has triggered,
+       do not congest the thread´s
+       message queue with IRQ events */
+    gpio_irq_disable(dev->params.pin_irq);
+    DEBUG("[nrf24l01p_ng] IRQ\n");
```

Comment:
```suggestion
    DEBUG_PUTS("[nrf24l01p_ng] IRQ");
```

Reply:
Yeah since it is a big wall of debug text, I should change this.

Btw. I have tested again with ENABLE_DEBUG (1)
and if I do `txtsnd 6 77:24:26:11:76 LOL`, on the other side I get
```
2020-05-04 19:41:39,774 # 0x77 0x24 0x26 0x11 0x76 0x4C 0x4F 0x4C 
2020-05-04 19:41:39,775 # [nrf24l01p_ng] Received frame length: 8
```
which is the format that is returned by `_recv()`